### ISSUE TITLE
[HOLD for payment 2024-03-29] [Performance] Chat Switch Improvement Part 1

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#32793]
Coming from [this proposal](https://expensify.slack.com/archives/C05LX9D6E07/p1702285001999819) from @hurali97 

### Problem/Solution:

On Android, switching a chat takes noticeable time and doesn’t offer a good user experience. Benchmarking the main on a Mid-range android device gives us the following metrics:

Main: 7616, 7628, 7639, 7643, 7649 = 7635 ms

These durations are recorded on a debug build so we can assume that in production it’ll be shorter but that doesn’t guarantee good user experience, especially on low-end/ mid-range device. After investigating, the areas we find that this is one of the most complex piece to improve, so let’s start with what’s rather doable.

LHNOptionList  leverages withCurrentReportID which adds to the rendering time and also re-rendering of the list when currentReportID is changed. The ideal solution is to put useCurrentReportID directly in the OptionRowLHNData since currentReportID is being used to evaluate isFocused prop for this component. We can save that process and do it where it’s needed to not render other parts of React tree.

When user taps on OptionRowLHNData the user navigates to the ReportScreen in that period a few unnecessary Onyx calls are being made:

First up is ComposeWithSuggestion where draft is set to false, even if it’s already set to false. Because in SilentCommentUpdater we call updateComment on mount.

Next, InteractionManager can be used to offload the work that happens before the user navigates to the ReportScreen. We can delay that work until the navigation is completed so that the user doesn’t feel that onPress is stuck.

Whole reportAction is being passed to ReportActionItem and it only uses some properties from it, so it makes sense to make it light weight and then pass it to ReportActionItem . The same concept is applied to ReportActionsList which now uses light weight report object.

Finally, there were handful unnecessary re-renders happening down the tree due to inline objects and functions :no_entry_sign:
After all these improvements following is the results on debug build on the mid-range android device:

Android: 4221, 4137, 4102, 4096, 4160 = 4143.2 ms

So there’s about 3500 ms reduction in the total time, giving us around 4143.2 ms ( as compared to 7635 ms ), from when user taps on the OptionRowLHN and until the report is visible. T[here](https://github.com/Expensify/App/pull/32336)’s still room for improvement in this area and we will try to follow up on this with another [PR](https://github.com/Expensify/App/pull/30168).
As of now, as part of the investigation we also created the PR here, which will be ready for review after some cleanup and failing tests 🙏 

P.S: some of the changes are already present in [PR](https://github.com/Expensify/App/pull/30168) so they will be removed from this PR once the mentioned PR is merged.
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b6dca257065e9f96</li>
        <li>Upwork Job ID: 1734155699026690048</li>
        <li>Last Price Increase: 2023-12-11</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alexpensify</details>

**Comment by mountiny** [comment]
Still working on this

**Comment by mountiny** [comment]
The PR had to be reverted

**Comment by alexpensify** [comment]
@allroundexperts is this warning an error or valid?

**Comment by allroundexperts** [comment]
This is valid. We have discovered a small UI issue.

**Comment by alexpensify** [comment]
I see one of the PR is merged into production. I think the automation is off here, it looks like there is another related PR in the review process. I think we should hold off on payments until then but keep me posted @allroundexperts.

**Comment by alexpensify** [comment]
@allroundexperts - we are all set here and there are no other PRs to wait for - right?

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/issues/38966#issuecomment-2019846492

Looks like this work, and the below PR might be responsible for this
- https://github.com/Expensify/App/pull/38255

Is this a bug that is already being considered and worked on? Or planned to be addressed in the future?

**Comment by alexpensify** [comment]
@allroundexperts - can we get an update for @mallenexpensify? I need to confirm if we should handle in another GH and continue with the payment process here. Thanks!

**Comment by mallenexpensify** [comment]
@allroundexperts 👀 on [the above plz](https://github.com/Expensify/App/issues/32793#issuecomment-2021487476)

**Comment by mountiny** [comment]
I think we can go ahead with payment on this one, the PR has been complicated and we already had second iteration on it. I have bumped the PR authors to look into that issue and review the proposals.

**Comment by alexpensify** [comment]
With @mountiny's feedback, here is the payment summary: 

-  External issue reporter - N/A 
-  Contributor that fixed the issue - @hurali97 payment handled by partner  
-  Contributor+ that helped on the issue and/or PR - @allroundexperts $500

Upwork Job: N/A 

Extra Notes regarding payment: There is regression which typically results in a decrease in pay. After reviewing this GH, these changes were complex. I'm suggesting leaving the bounty amount at the standard rate.

**Comment by mallenexpensify** [comment]
@alexpensify , added `is due payment via NewDot,` to make it clear I have the below saved as a textexpander, to help keep things consistent. 

Contributor: @ paid $ via Upwork
Contributor+: @ paid $ via NewDot

**Comment by alexpensify** [comment]
Noted!

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts
